lib/index.js

```diff
@@ -64,9 +100,9 @@ export default class Pageres {
 	 * @api public
 	 */
 
-	async run() {
+	async run(): Promise<stream$Readable[]> { // TODO: any way to define that it returns `this.items` and use its type?
```

Comment:
Do I add the `<T: Item>` to the existing `Pageres` class? I ask because in the other example you added `class Foo<Value>` to the class and I'm not really sure what these mean or how to do both.

And your example isn't directly applicable as in the actual code here, `this.items` is just assigned with an empty array in the constructor, not a passed value.

What I need to denote is that `async method()` returns the `items: T;` definition. Not where it cam from.